File: Group B/Code/cpp/resize.cpp
#include <iostream>
#include <opencv2/opencv.hpp>
#include <opencv2/highgui/highgui.hpp>
#include <opencv2/imgproc/imgproc.hpp>
#include <opencv2/core/core.hpp>
#include <opencv2/highgui/highgui.hpp>
#include <vector>
#include <string>
#include <filesystem>

namespace fs = std::filesystem;

int main() {
    std::string source_directory = "../../opencv_dataset/yolo_objects";
    std::string destination_directory = "resizeDone";
    int target_width = 1600;
    int target_height = 1200;

    double start_time = cv::getTickCount();

    for (const auto& entry : fs::directory_iterator(source_directory)) {
        std::string filename = entry.path().filename().string();
        std::string extension = fs::path(filename).extension().string();
        if (extension == ".jpg" || extension == ".jpeg" || extension == ".png" || extension == ".bmp" || extension == ".gif") {
            cv::Mat image = cv::imread(entry.path().string());

            if (!image.empty()) {
                cv::Mat resized_image;
                cv::resize(image, resized_image, cv::Size(target_width, target_height));

                std::string destination_path = destination_directory + "/" + filename;
                cv::imwrite(destination_path, resized_image);

                std::cout << "Resized and saved: " << destination_path << std::endl;
            }
        }
    }

    double end_time = cv::getTickCount();
    double execution_time = (end_time - start_time) / cv::getTickFrequency();

    std::cout << "Execution Time: " << execution_time << " seconds" << std::endl;

    return 0;
}

File: Group B/Code/python/resize.py
import os
import cv2

source_directory = '../opencv_dataset/yolo_objects'
destination_directory = 'resizeDone'
target_width = 1600
target_height = 1200
files = os.listdir(source_directory)

start_time = cv2.getTickCount()

for filename in files:
    if filename.endswith(('.jpg', '.jpeg', '.png', '.bmp', '.gif')):
        image = cv2.imread(os.path.join(source_directory, filename))
        
        if image is not None:
            resized_image = cv2.resize(image, (target_width, target_height))
            destination_path = os.path.join(destination_directory, filename)
            cv2.imwrite(destination_path, resized_image)
            
            print(f'Resized and saved: {destination_path}')

end_time = cv2.getTickCount()
execution_time = (end_time - start_time) / cv2.getTickFrequency()

print(f"Execution Time: {execution_time} seconds")
